From e58f6aa323e1d29604ed5c68bacfef0e4ef9d1f2 Mon Sep 17 00:00:00 2001
From: 115ek <e.blosz@hotmail.de>
Date: Tue, 18 May 2021 21:12:47 +0200
Subject: [PATCH] Settings: Relax conditions for NFC settings

The if condition is wrong here. If a device only features
FEATURE_NFC but not FEATURE_NFC_HOST_CARD_EMULATION,
the nfc entry will be missing from the settings menu.

As we check HCE in an own controller, drop it here.

Change-Id: I977eac4b59f03f61b6a29b512ff5eb5ba390f066
---
 .../connecteddevice/NfcAndPaymentFragmentController.java      | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/com/android/settings/connecteddevice/NfcAndPaymentFragmentController.java b/src/com/android/settings/connecteddevice/NfcAndPaymentFragmentController.java
index 65fafee0d1..7f140f9b71 100644
--- a/src/com/android/settings/connecteddevice/NfcAndPaymentFragmentController.java
+++ b/src/com/android/settings/connecteddevice/NfcAndPaymentFragmentController.java
@@ -77,9 +77,7 @@ public class NfcAndPaymentFragmentController extends BasePreferenceController
 
     @Override
     public int getAvailabilityStatus() {
-        if (!mPackageManager.hasSystemFeature(PackageManager.FEATURE_NFC)
-                || !mPackageManager.hasSystemFeature(
-                PackageManager.FEATURE_NFC_HOST_CARD_EMULATION)) {
+        if (!mPackageManager.hasSystemFeature(PackageManager.FEATURE_NFC)) {
             return UNSUPPORTED_ON_DEVICE;
         }
         return AVAILABLE;
-- 
2.20.1

